File: P5 2024-2/Paradigmas de Programacao/ListaExercicio01/Golang/Imprimir.go
package main

import "fmt"

type imprimivel interface {
	imprimir()
}

type Relatorio struct {
	corpo string
	imprimivel
}

type Contrato struct {
	corpo       string
	contratante string
	imprimivel
}

func (r *Relatorio) imprimir() {
	fmt.Println("Imprimindo relatorio: \n", r.corpo)
}

func (c *Contrato) imprimir() {
	fmt.Println("Imprimindo contrato de", c.contratante, "\n", c.corpo)
}

File: P5 2024-2/Paradigmas de Programacao/ListaExercicio01/Python/Imprimir.py
class Imprimivel:
    def imprimir(self):
        pass

class Relatorio(Imprimivel):
    def __init__(self, corpo):
        self.corpo = corpo

    def imprimir(self):
        print(f"Imprimindo Relatório: {self.corpo}")

class Contrato(Imprimivel):
    def __init__(self, corpo, contratante):
        self.corpo = corpo
        self.contratante = contratante

    def imprimir(self):
        print(f"Imprimindo Contrato de {self.contratante}: {self.corpo};")